lib/compat/wordpress-6.5/rest-api.php

```diff
@@ -19,3 +19,91 @@ function gutenberg_register_global_styles_revisions_endpoints() {
 }
 
 add_action( 'rest_api_init', 'gutenberg_register_global_styles_revisions_endpoints' );
+
+
+/**
+ * Updates REST API response for the attachments and adds extra properties.
+ *
+ * @param WP_REST_Response $response The response object.
+ * @param WP_Post          $post    The original attachment post.
+ * @return WP_REST_Response $response Updated response object.
+ */
+function gutenberg_modify_rest_attachments_response( $response, $post ) {
+	$uncompressed_media_id = get_post_meta( $post->ID, '_wp_uncompressed_media_id', true );
+	if ( $uncompressed_media_id ) {
+		$response->data['uncompressed_source_url'] = wp_get_attachment_url( $uncompressed_media_id );
+	}
+	// There is a cached value for the file size, but it seems that it's
+	// not reliable (@see https://core.trac.wordpress.org/ticket/57459).
+	$response->data['filesize'] = filesize( get_attached_file( $post->ID ) );
```

Comment:
File size information is already stored in post meta and is part of the rest api response. Please don't add this. 

Also use wp_filesize

Reply:
I mention in the code comment that this value is not reliable and might be [removed from core](https://core.trac.wordpress.org/ticket/57459).